internal/job/repository_job.go

```diff
@@ -0,0 +1,207 @@
+			if err != nil {
+				return errors.Wrap(err, op)
+			}
+			return nil
+		},
+	)
+
+	if err != nil {
+		if errors.IsUniqueError(err) {
+			return nil, errors.Wrap(err, op, errors.WithMsg(fmt.Sprintf("name %s already exists", j.Name)))
+		}
+		return nil, errors.Wrap(err, op)
+	}
+	return j, nil
+}
+
+// UpdateJob updates the repository entry for j.PrivateId with the values
+// in j for the fields listed in fieldMaskPaths. It returns a new Job
+// containing the updated values and a count of the number of records
+// updated. j is not changed.
+//
+// j must contain a valid PrivateId. Only j.Description, and
+// j.NextScheduledRun can be updated.
+//
+// All options are ignored.
+func (r *Repository) UpdateJob(ctx context.Context, j *Job, fieldMaskPaths []string, _ ...Option) (*Job, int, error) {
+	const op = "job.(Repository).UpdateJob"
+	if j == nil {
+		return nil, db.NoRowsAffected, errors.New(errors.InvalidParameter, op, "missing job")
+	}
+	if j.Job == nil {
+		return nil, db.NoRowsAffected, errors.New(errors.InvalidParameter, op, "missing embedded job")
+	}
+	if j.PrivateId == "" {
+		return nil, db.NoRowsAffected, errors.New(errors.InvalidParameter, op, "missing private id")
+	}
+
+	for _, f := range fieldMaskPaths {
+		switch {
+		case strings.EqualFold("Description", f):
+		case strings.EqualFold("NextScheduledRun", f):
+		default:
+			return nil, db.NoRowsAffected, errors.New(errors.InvalidFieldMask, op, fmt.Sprintf("invalid field mask: %s", f))
+		}
+	}
+	var dbMask, nullFields []string
+	dbMask, nullFields = common.BuildUpdatePaths(
+		map[string]interface{}{
+			"Description":      j.Description,
+			"NextScheduledRun": j.NextScheduledRun,
+		},
+		fieldMaskPaths,
+		nil,
+	)
+	if len(dbMask) == 0 && len(nullFields) == 0 {
+		return nil, db.NoRowsAffected, errors.New(errors.EmptyFieldMask, op, "empty field mask")
+	}
+
+	oplogWrapper, err := r.kms.GetWrapper(ctx, scope.Global.String(), kms.KeyPurposeOplog)
+	if err != nil {
+		return nil, db.NoRowsAffected, errors.Wrap(err, op, errors.WithMsg("unable to get oplog wrapper"))
+	}
+
+	var rowsUpdated int
+	j = j.clone()
+	_, err = r.writer.DoTx(ctx, db.StdRetryCnt, db.ExpBackoff{},
+		func(_ db.Reader, w db.Writer) error {
+			var err error
+			rowsUpdated, err = w.Update(ctx, j, dbMask, nullFields, db.WithOplog(oplogWrapper, j.oplog(oplog.OpType_OP_TYPE_UPDATE)))
+			if err != nil {
+				return errors.Wrap(err, op)
+			}
+			if rowsUpdated > 1 {
+				return errors.New(errors.MultipleRecords, op, "more than 1 resource would have been updated")
+			}
+			return nil
+		},
+	)
+	if err != nil {
+		return nil, db.NoRowsAffected, errors.Wrap(err, op)
+	}
+
+	return j, rowsUpdated, nil
+}
+
+// LookupJob will look up a job in the repository using the privateId. If the job is not
+// found, it will return nil, nil.
+//
+// All options are ignored.
+func (r *Repository) LookupJob(ctx context.Context, privateId string, _ ...Option) (*Job, error) {
```

Comment:
Will a `ListJobs` method be added?

Reply:
I am not sure, I had a simple `ListJob` but I didnt really have a use case for it, so I pulled it out